xalia/Win32/Win32Connection.cs

```diff
@@ -112,5 +257,82 @@ public string[] GetTrackedProperties()
+
+            if (new_threadinfo.hwndActive != old_threadinfo.hwndActive)
+            {
+                Root.PropertyChanged("win32_gui_hwndactive", new_threadinfo.hwndActive);
+                LookupElement(old_threadinfo.hwndActive)?.PropertyChanged("win32_gui_active", "false");
+                LookupElement(new_threadinfo.hwndActive)?.PropertyChanged("win32_gui_active", "true");
+            }
+
+            if (new_threadinfo.hwndFocus != old_threadinfo.hwndFocus)
+            {
+                Root.PropertyChanged("win32_gui_hwndfocus", new_threadinfo.hwndFocus);
+                LookupElement(old_threadinfo.hwndFocus)?.PropertyChanged("win32_gui_focus", "false");
+                LookupElement(new_threadinfo.hwndFocus)?.PropertyChanged("win32_gui_focus", "true");
+            }
+
+            if (new_threadinfo.hwndCapture != old_threadinfo.hwndCapture)
+            {
+                Root.PropertyChanged("win32_gui_hwndcapture", new_threadinfo.hwndCapture);
+                LookupElement(old_threadinfo.hwndCapture)?.PropertyChanged("win32_gui_capture", "false");
+                LookupElement(new_threadinfo.hwndCapture)?.PropertyChanged("win32_gui_capture", "true");
+            }
+
+            if (new_threadinfo.hwndMenuOwner != old_threadinfo.hwndMenuOwner)
+            {
+                Root.PropertyChanged("win32_gui_hwndmenuowner", new_threadinfo.hwndMenuOwner);
+                LookupElement(old_threadinfo.hwndMenuOwner)?.PropertyChanged("win32_gui_menuowner", "false");
+                LookupElement(new_threadinfo.hwndMenuOwner)?.PropertyChanged("win32_gui_menuowner", "true");
+            }
+
+            if (new_threadinfo.hwndMoveSize != old_threadinfo.hwndMoveSize)
+            {
+                Root.PropertyChanged("win32_gui_hwndmovesize", new_threadinfo.hwndMoveSize);
+                LookupElement(old_threadinfo.hwndMoveSize)?.PropertyChanged("win32_gui_movesize", "false");
+                LookupElement(new_threadinfo.hwndMoveSize)?.PropertyChanged("win32_gui_movesize", "true");
+            }
+
+            // We don't have a way to get caret change notifications so we'll ignore that for now
+        }
+
+        private void OnMsaaEvent(IntPtr hWinEventProc, uint eventId, IntPtr hwnd, int idObject, int idChild, int idEventThread, int dwmsEventTime)
+        {
+            switch (eventId)
+            {
+                case EVENT_SYSTEM_FOREGROUND:
+                case EVENT_SYSTEM_MENUSTART:
+                case EVENT_SYSTEM_MENUEND:
+                case EVENT_SYSTEM_MENUPOPUPSTART:
+                case EVENT_SYSTEM_MENUPOPUPEND:
+                case EVENT_SYSTEM_CAPTURESTART:
+                case EVENT_SYSTEM_CAPTUREEND:
+                case EVENT_SYSTEM_MOVESIZESTART:
+                case EVENT_SYSTEM_MOVESIZEEND:
+                case EVENT_OBJECT_FOCUS:
+                    UpdateGuiThreadInfo();
+                    break;

```

Comment:
Not sure how much overhead is associated with this update, but might only be worth doing if idChild is `CHILDID_SELF`. Otherwise, you're likely to end up updating the GuiThreadInfo for child elements that don't have their own HWND.

Reply:
GetGUIThreadInfo is a user32 call, so I expect it to cost 1 syscall. The rest is comparing fields in a struct, so it should also have little cost except when things change.